source/common/json/json_loader.cc

```diff
@@ -521,7 +523,7 @@ bool ObjectHandler::StartObject() {
     state_ = expectKeyOrEndObject;
     return true;
   default:
```

Comment:
we should delete `default` completely from the switch() as it should never happen.

the compiler will enforce that all types of state_ are present in the `case` statements. (could place NOT_REACHED at the very end after switch)